py/dml/messages.py

```diff
@@ -1138,6 +1138,25 @@ class EAUTOPARAM(DMLError):
     library, and they may not be overridden."""
     fmt = "bad declaration of automatic parameter '%s'"
 
+class ENWALRUS(DMLError):
+    """When the `walrus_params` provisional feature is enabled, parameter
+    declarations that do not override existing declarations must be
+    declared using the `:=` or `:default` syntax.
+    """
+    fmt = "non-overriding param should use :%s syntax (from walrus_params feature)"
```

Comment:
This error message is the **key feature** of walrus params, and the message should play to that.
```
fmt = "parameter not previously declared: '%s'. To declare and define a new parameter, use the ':%s' syntax (this requirement is due to the 'walrus_params' provisional feature being enabled)."
```

Reply:
Improved. I chose to include the provisional declaration on a separate context line.